app/controllers/provider/admin/dashboard/widget_base_controller.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+class Provider::Admin::Dashboard::WidgetBaseController < Provider::Admin::BaseController
+  include DashboardTimeRange
+
+  helper_method :current_range, :previous_range, :widget
+
+  respond_to :js
+
+  def show
+    render 'provider/admin/dashboard/widget/show'
+  end
+
+  def widget
+    @widget ||= presenter.new(widget_data)
```

Comment:
If `presenter` and `widget_data` are required to be defined in subclasses, they should be defined in this parent class as empty methods that raise `NoMethodError`.